employee-rostering-backend/src/test/java/org/optaweb/employeerostering/contract/ContractServiceTest.java

```diff
@@ -103,23 +106,23 @@ public void getNonExistentContractTest() {
 
     @Test
     public void getNonMatchingContractTest() {
-        Integer tenantId = 1;
+        Integer tenantId = 2;
```

Comment:
Why did you need to change the tenantId?

Reply:
This is a temporary hack to get around the fact that `RosterGenerator` generates entities whose `id`'s interfere with the entities' `id`'s in the test classes (one of the Tenants created by `RosterGenerator` has `id = 1`, which is the same `id` used in this test). Once I implement Tenant CRUD I will add `createTestTenant()` methods to each test class so that each test class has its own persisted `Tenant` entity that will not interfere with `RosterGenerator`. This . Your `BaseTest` solution also gets around this problem, but with this alternative we don't have to "turn off" `RosterGenerator` for certain test classes. I'm open to discussing this, love to know what you think.